### PR TITLE
Changed GetEmployeeTeamSnapshot call to match new schema

### PR DESCRIPTION
**Pull request by nmcmahon-unosquare**
Created new query.
Added new models
Updated existing controller, mappers and models.

### CODE REVIEW

File: v2/AdminCore.Services/DashboardService.cs [hunk 1]
```diff
@@ -163,14 +165,64 @@ private static EmployeeSnapshotDto CreateEmployeeSnapshotFromContract(Contract c
...
+        DateService.EventContainsEventDatesThatHappenDuringAGivenDay(evnt.EventDates, date)).ToList();
+      return employeeEvents.Any() ? GetLocationFromEvent(employeeEvents.First()) : EmployeeLocationConstants.InOffice;
+    }
+
+    private static string GetLocationFromEvent(Event @event)
+    {
+      switch (@event.EventTypeId)
+      {
+        case 1:
+          return EmployeeLocationConstants.OnHoliday;
```
Comment (Jamie-Unosquare) [inline v2/AdminCore.Services/DashboardService.cs:216]: Small thing, change this to AnnualLeave for consistency with the rest of the project

File: v2/AdminCore.Constants/EmployeeLocationConstants.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AdminCore.Constants
+{
+  public class EmployeeLocationConstants
+  {
+    public static string OnHoliday = "On Holiday";
```
Comment (Jamie-Unosquare) [inline v2/AdminCore.Constants/EmployeeLocationConstants.cs:9]: Small thing, change this to AnnualLeave for consistency with the rest of the project

File: v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs [hunk 3]
```diff
@@ -146,7 +146,7 @@ public void GetTeamEvents_WhenCalled_ReturnsTeamEvents()
       var eventsReturnedFromService = _fixture.CreateMany<EventDto>(numberOfEvents).ToList();
       var dashboardEventModels = _fixture.CreateMany<DashboardEventViewModel>(numberOfEvents);
 
-      _dashboardService.GetTeamDashboardEvents(Arg.Is(employeeId), searchDate).Returns(eventsReturnedFromService);
+      //_dashboardService.GetTeamDashboardEvents(Arg.Is(employeeId), searchDate).Returns(eventsReturnedFromService);
```
Comment (Jamie-Higgins) [inline v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs]: Excuse me?!

File: v2/AdminCore.WebApp/Mappings/WebMappingProfile.cs [hunk 4]
```diff
@@ -69,6 +71,14 @@ public WebMappingProfile()
       CreateMap<ContractDto, ContractViewModel>().ReverseMap();
       CreateMap<ContractDto, CreateContractViewModel>().ReverseMap();
       CreateMap<ContractDto, UpdateContractViewModel>().ReverseMap();
+
+      CreateMap<ClientSnapshotDto, ClientSnapshotViewModel>();
```
Comment (Jamie-Unosquare) [inline v2/AdminCore.WebApp/Mappings/WebMappingProfile.cs]: These 3 are unnecessary as ReverseMap includes the mapping for both directions

File: v2/AdminCore.Services/Mappings/DashboardMapperProfile.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+using AdminCore.DTOs.Dashboard;
+using AutoMapper;
+
+namespace AdminCore.Services.Mappings
+{
+  public class DashboardMapperProfile : Profile
+  {
+    public DashboardMapperProfile()
+    {
+      CreateMap<Client, ClientSnapshotDto>();
```
Comment (Jamie-Unosquare) [inline v2/AdminCore.Services/Mappings/DashboardMapperProfile.cs]: These 3 are unnecessary as ReverseMap includes the mapping for both directions

**Review by Jamie-Unosquare — CHANGES_REQUESTED** [review]
Change constant name and remove unused mappings

File: v2/AdminCore.Services/DashboardService.cs (hunk 1 above)
Comment (nmcmahon-unosquare) [reply v2/AdminCore.Services/DashboardService.cs:216]: OnHoliday encompasses multiple holiday types (Annual Leave, Public Holiday, Mandatory leave.. etc..) and was explicitly requested by FE

**Review by Jamie-Unosquare — APPROVED** [review]

**Pull request merged**